src/app/Notifications/Notifications.tsx

```diff
@@ -60,17 +60,17 @@ export class Notifications {
   );
   private readonly _drawerState$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
 
-  constructor() {
-    this._drawerState$
-      .pipe(filter((v) => v))
-      .subscribe(() => this._notifications$.next(this._notifications.map((n) => ({ ...n, hidden: true }))));
```

Comment:
> Looks like I tried to get too fancy with the observables.

XD I think it was working if we modify the notification in-place....Pretty odd.


```typescript
this._drawerState$
      .pipe(filter((v) => v))
      .subscribe(() => this._notifications$.next(this._notifications.map((n) => { 
        n.hidden = true;
        return n;
      })));
```

Reply:
... alright then. :-/

Reply:
Ahhh I think here we call `next` but that new array is not assigned to internal `this._notifications` array, so it was using the old one this while time.